File: solutions/csharp/jedliks-toys/1/ElonsToys.cs
using System;

class RemoteControlCar
{
    private int _metersDriven = 0;
    private int _batteryCharge = 100;
    
    public static RemoteControlCar Buy() => new RemoteControlCar();

    public string DistanceDisplay() => $"Driven {_metersDriven} meters";

    public string BatteryDisplay() => _batteryCharge > 0 ? $"Battery at {_batteryCharge}%" : "Battery empty";

    public void Drive()
    {
        if (_batteryCharge > 0)
        {
            --_batteryCharge;
            _metersDriven += 20;
        }
    }
}

File: solutions/go/jedliks-toys/1/elons_toys.go
package elon

import "fmt"

func (c *Car) Drive() {
    if newBattery := c.battery - c.batteryDrain; newBattery >= 0 {
        c.battery = newBattery
        c.distance += c.speed
    }
}

func (c Car) DisplayDistance() string {
    return fmt.Sprintf("Driven %d meters", c.distance)
}

func (c Car) DisplayBattery() string {
    return fmt.Sprintf("Battery at %d%%", c.battery)
}

func (c Car) CanFinish(trackDistance int) bool {
    if trackDistance == 0 {
        return true
    }
    return (((trackDistance - 1) / c.speed) + 1) * c.batteryDrain <= c.battery
}
